### PR TITLE
fix: batch changed_book_uuids annotation lookup (#1446)

### PR DESCRIPTION
**Pull request by roberte777**
## Summary
- Closes #1446
- `changed_book_uuids` (the query backing the Kobo device's `POST /api/v3/content/checkforchanges` polling heartbeat) ran one `served_kobo_annotations` query per candidate book instead of a single batched query.
- Added `served_kobo_annotations_batch` next to `served_kobo_annotations` in `db/src/annotations/mod.rs`: one `WHERE book_uuid IN (...)` query (chunked at 900 to stay under SQLite's 999 bind-parameter cap), grouped by `book_uuid` into a `HashMap<String, Vec<ServedKoboAnnotation>>`.
- `changed_book_uuids` in `db/src/kobo/annotations.rs` now fetches all candidates in one batch call and iterates the returned map to compute fingerprints and the adopted/acked comparison exactly as before — mirrors `sync_delta`'s existing use of the batched `reading_state_for` in the same feature area (`db/src/kobo/delta.rs`).
- No change to `checkforchanges`'s response shape; all existing `changed_book_uuids` test expectations are unchanged.

## Test plan
- [x] `cargo fmt` — clean
- [x] `cargo clippy -p omnibus-db --all-targets -- -D warnings` — clean, no warnings
- [x] `cargo test -p omnibus-db` — `test result: ok. 1471 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out`
  - Includes a new test, `changed_book_uuids_computes_the_correct_set_across_multiple_candidate_books`, exercising the batched path with 3 candidate books in three different states (acked-and-quiet, never-acked, stale-ack-after-a-recolor).
  - Also added direct unit tests for `served_kobo_annotations_batch` (grouping/omission behavior, empty-input, db-error propagation) in `db/src/annotations/tests.rs`.
  - All pre-existing `changed_book_uuids` tests pass unmodified.

## Version
- [x] **Patch** — the default; leaving unlabeled per the template (internal tech-debt fix, no user-facing behavior change).

## Notes
- Chunk size 900 leaves headroom under SQLite's 999-bind cap (one bind per candidate uuid plus `user_id` per chunk), matching the chunking convention used by `reading_state_for` and `resolve_canonical_book_uuids_bulk_exec` elsewhere in the crate.
- The batch helper does not re-resolve merged/canonical uuids (unlike the single-uuid `served_kobo_annotations`) — every caller of `changed_book_uuids` already canonicalizes `book_uuid` before it reaches `kobo_annotations_sync`/`annotations`, so this preserves behavior while avoiding an unnecessary per-uuid resolve.

---
_Generated by [Claude Code](https://claude.ai/code/session_01VPCQcEtXzv2WJg7g4Ci3Rn)_

### CODE REVIEW

File: db/src/annotations/mod.rs [hunk 1]
```diff
@@ -284,6 +284,60 @@ pub async fn served_kobo_annotations(
...
+               FROM annotations
+              WHERE user_id = ? AND book_uuid IN ({placeholders})
+                AND kobo_location IS NOT NULL AND client_id IS NOT NULL
+              ORDER BY created_at ASC"
+        );
+        let mut q = sqlx::query(&sql).bind(user_id);
+        for uuid in chunk {
+            q = q.bind(uuid);
+        }
+        for row in q.fetch_all(pool).await? {
```
[the remark this answers is not in the document]
Comment (roberte777) [reply db/src/annotations/mod.rs:334]: Good catch, confirmed real. `served_kobo_annotations_batch` now applies the same `LIST_HIGHLIGHTS_LIMIT` per book via a `ROW_NUMBER() OVER (PARTITION BY book_uuid ORDER BY created_at ASC)` window filtered to `rn <= LIST_HIGHLIGHTS_LIMIT`, mirroring `covers_manual_batch`'s windowed pattern in `db/src/shelves/read.rs`. That keeps the batch's per-book row set/order identical to what `served_kobo_annotations` returns for that book, so the fingerprint a device acked from `GET /annotations` can match the one `checkforchanges` recomputes again. Doc comment updated too. Fixed in 5a12102.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: db/src/annotations/tests.rs [hunk 2]
```diff
@@ -682,6 +682,84 @@ async fn served_kobo_annotations_returns_empty_for_an_unknown_book() {
...
+
+#[tokio::test]
+async fn served_kobo_annotations_batch_returns_empty_map_for_no_candidates() {
+    let pool = init_db("sqlite::memory:").await.unwrap();
+    let user = seed_user(&pool, "alice").await;
+    assert!(served_kobo_annotations_batch(&pool, user, &[])
+        .await
+        .unwrap()
+        .is_empty());
+}
```
[the remark this answers is not in the document]
Comment (roberte777) [reply db/src/annotations/tests.rs:796]: Added `served_kobo_annotations_batch_caps_each_book_at_list_highlights_limit_like_the_single_uuid_form` in `db/src/annotations/tests.rs`: seeds `LIST_HIGHLIGHTS_LIMIT + 5` Kobo-placeable rows for one book (explicit strictly-increasing `created_at` to avoid same-second ties), then asserts both `served_kobo_annotations` and `served_kobo_annotations_batch` return exactly `LIST_HIGHLIGHTS_LIMIT` rows for that book and that their fingerprints match — which fails without the fix and passes with it. Fixed in 5a12102.

---
_Generated by [Claude Code](https://claude.ai/code)_
